astroquery/mast/collections.py

```diff
@@ -0,0 +1,504 @@
+    """
+
+    def __init__(self):
+
+        super().__init__()
+
+        services = {"panstarrs": {"path": "panstarrs/{data_release}/{table}.json",
+                                  "args": {"data_release": "dr2", "table": "mean"}}}
+        self._service_api_connection.set_service_params(services, "catalogs", True)
+
+        self.catalog_limit = None
+        self._current_connection = None
+
+    def _parse_result(self, response, verbose=False):
+
+        results_table = self._current_connection._parse_result(response, verbose)
+
+        if len(results_table) == self.catalog_limit:
+            warnings.warn("Maximum catalog results returned, may not include all sources within radius.",
+                          MaxResultsWarning)
+
+        return results_table
+
+    @class_or_instance
+    def query_region_async(self, coordinates, radius=0.2*u.deg, catalog="Hsc",
+                           version=None, pagesize=None, page=None, **kwargs):
+        """
+        Given a sky position and radius, returns a list of catalog entries.
+        See column documentation for specific catalogs `here <https://mast.stsci.edu/api/v0/pages.htmll>`__.
+
+        Parameters
+        ----------
+        coordinates : str or `~astropy.coordinates` object
+            The target around which to search. It may be specified as a
+            string or as the appropriate `~astropy.coordinates` object.
+        radius : str or `~astropy.units.Quantity` object, optional
+            Default 0.2 degrees.
+            The string must be parsable by `~astropy.coordinates.Angle`. The
+            appropriate `~astropy.units.Quantity` object from
+            `~astropy.units` may also be used. Defaults to 0.2 deg.
+        catalog : str, optional
+            Default HSC.
+            The catalog to be queried.
+        version : int, optional
+            Version number for catalogs that have versions. Default is highest version.
+        pagesize : int, optional
+            Default None.
+            Can be used to override the default pagesize for (set in configs) this query only.
+            E.g. when using a slow internet connection.
+        page : int, optional
+            Default None.
+            Can be used to override the default behavior of all results being returned to
+            obtain a specific page of results.
+        **kwargs
+            Other catalog-specific keyword args.
+            These can be found in the (service documentation)[https://mast.stsci.edu/api/v0/_services.html]
+            for specific catalogs. For example one can specify the magtype for an HSC search.
+
+        Returns
+        -------
+        response : list of `~requests.Response`
+        """
+
+        # Put coordinates and radius into consistant format
+        coordinates = commons.parse_coordinates(coordinates)
+
+        # if radius is just a number we assume degrees
+        if isinstance(radius, (int, float)):
+            radius = radius * u.deg
```

Comment:
not sure the `isinstance` check is great here; maybe `radius = u.Quantity(radius, u.deg)` would work?  Then, no matter what data type is input - string, int, float, array - we'll get the right thing out.

Reply:
Same idea, but one less line of code, adding a unit of `u.deg` to the `coord.Angle` call on the next line takes care of it. 😄